tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+from ocp_resources.network_policy import NetworkPolicy
+from ocp_resources.pod import Pod
+from pyhelper_utils.shell import run_command
+
+pytestmark = [pytest.mark.post_upgrade, pytest.mark.sno, pytest.mark.arm64, pytest.mark.s390x]
+
+
+cnv_network_policy_parametrize = pytest.mark.parametrize(
+    "component_name,port",
+    [
+        (
+            "kubevirt-console-plugin",
+            9443,
+        ),
+        (
+            "kubevirt-apiserver-proxy",
+            8080,
+        ),
+    ],
+)
+
+
+@pytest.fixture(scope="module")
+def cnv_network_policies(admin_client, hco_namespace):
+    """Get all network policies in the CNV namespace"""
+    return list(NetworkPolicy.get(dyn_client=admin_client, namespace=hco_namespace.name))
+
+
+@pytest.fixture
+def component_pods_for_test(admin_client, hco_namespace, component_name):
+    """Get component pods and return source/target pod info for connectivity testing"""
+
+    def _get_component_pods(admin_client, namespace, component_label, component_name):
+        """Helper function to get CNV component pods with their IPs"""
+        # Build full label selector with common CNV label + component-specific label
+        full_label_selector = f"app=kubevirt-hyperconverged,{component_label}"
```

Comment:
Use directly and remove the argument component_label.
```suggestion
        full_label_selector = f"app=kubevirt-hyperconverged, app.kubernetes.io/component={component_name}"
```

Reply:
I refactored this and now the test is getting the label selector from the NetworkPolicy. Please take a look to the new implementation and let me know any comment. Thanks!